File: core/RemoveElement/RemoveElement.py
from typing import List


def removeElement(nums: List[int], val: int) -> int:
    point, cur, length = 0, 0, len(nums)
    while cur < length:
        if nums[cur] != val:
            nums[cur], nums[point] = nums[point], nums[cur]
            point += 1
        cur += 1
    return point

File: core/RemoveElement/RemoveElement.java
package com.dbc;

public class RemoveElement {
    public int removeElement(int[] nums, int val) {
        int point = 0, cur = 0;
        while (cur < nums.length) {
            if (nums[cur] != val) {
                int temp = nums[cur];
                nums[cur] = nums[point];
                nums[point] = temp;
                point++;
            }
            cur++;
        }
        return point;
    }
}
